python/pyarrow/dataset.py

```diff
@@ -517,10 +517,11 @@ def dataset(source, schema=None, format=None, filesystem=None,
         fashion. Disabling this feature will skip the IO, but unsupported
         files may be present in the Dataset (resulting in an error at scan
         time).
-    ignore_prefixes : list, optional
+    selector_ignore_prefixes : list, optional
```

Comment:
Personally, I would keep this as `ignore_prefixes` in the public API. "selector" is a strange thing for users, as they don't necessarily pass that (rather a string path to a directory). I think a note in the explanation that this is only applicable for directories (as you did) is good enough.

Reply:
Agreed.